components/cluster/command/scale_out.go

```diff
@@ -247,11 +248,50 @@ func buildScaleOutTask(
 				filepath.Join(deployDir, "conf"),
 				filepath.Join(deployDir, "scripts")).
 			Mkdir(metadata.User, inst.GetHost(), dataDirs...)
+
+		// copy dependency component if needed
+		switch inst.ComponentName() {
```

Comment:
IMO, the L253-L294 can be extracted as a function(don't repeat yourself).

Reply:
They should not be in `deploy.go` from the very begaining, they should be handled in manifest's dependency logic. But we don't have that implemented now. It's ok to put these to a dedicated task like `DeploySpark` or so, but I do this in `deploy.go` for I don't expect these lines to be here for long.